test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildServerIntegrationTest.cs

```diff
@@ -143,28 +144,55 @@ public async Task Build_ServerConnectionMutexCreationFails_FallsBackToInProcessR
             // it reaches server creation part.
         }
 
-        // Skipping on linux/mac because of https://github.com/aspnet/Razor/issues/2507.
+        // Skipping on MacOS because of https://github.com/dotnet/corefx/issues/33141.
         [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
         [InitializeTestProject("SimpleMvc")]
         public async Task ManualServerShutdown_NoPipeName_ShutsDownServer()
         {
-            var toolAssembly = typeof(Application).Assembly.Location;
-            var result = await DotnetMSBuild(
-                "Build",
-                $"/p:_RazorForceBuildServer=true /p:_RazorToolAssembly={toolAssembly}",
-                suppressBuildServer: true); // We don't want to specify a pipe name
+            // Publish rzc tool to a temporary path
```

Comment:
Can you add a comment to say why you're doing this? E.g. the pipe name is calculated from the path of the dll